File: src/main/java/no/nav/pensjon/innsyn/poi/ss/RowFiller.java
package no.nav.pensjon.innsyn.poi.ss;

import no.nav.pensjon.innsyn.sink.CellValueSetter;
import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.CellStyle;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;

import java.util.Date;

public class RowFiller implements CellValueSetter {

    private final CellStyle dateCellStyle;
    private final Row row;

    RowFiller(Sheet sheet, CellStyle dateCellStyle, int rowIndex) {
        this.dateCellStyle = dateCellStyle;
        this.row = sheet.createRow(rowIndex);
    }

    public void setCellValue(int cellIndex, double value) {
        row.createCell(cellIndex).setCellValue(value);
    }

    public void setCellValue(int cellIndex, String value) {
        row.createCell(cellIndex).setCellValue(value);
    }

    public void setCellValue(int cellIndex, Date value) {
        createDateCell(cellIndex).setCellValue(value);
    }

    private Cell createCell(int cellIndex) {
        return row.createCell(cellIndex);
    }

    private Cell createDateCell(int cellIndex) {
        Cell cell = createCell(cellIndex);
        cell.setCellStyle(dateCellStyle);
        return cell;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/poi/ss/RowFiller.kt
package no.nav.pensjon.innsyn.poi.ss

import no.nav.pensjon.innsyn.domain.map.CellValueSetter
import org.apache.poi.ss.usermodel.CellStyle
import org.apache.poi.ss.usermodel.Sheet
import java.util.*

class RowFiller internal constructor(sheet: Sheet, private val dateCellStyle: CellStyle, rowIndex: Int) : CellValueSetter {
    private val row = sheet.createRow(rowIndex)
    override fun setCellValue(cellIndex: Int, value: Boolean) {
        row.createCell(cellIndex).setCellValue(if (value) "ja" else "nei")
    }

    override fun setCellValue(cellIndex: Int, value: Double) {
        row.createCell(cellIndex).setCellValue(value)
    }

    override fun setCellValue(cellIndex: Int, value: String) {
        row.createCell(cellIndex).setCellValue(value)
    }

    override fun setCellValue(cellIndex: Int, value: Date) {
        createDateCell(cellIndex).setCellValue(value)
    }

    private fun createCell(cellIndex: Int) = row.createCell(cellIndex)

    private fun createDateCell(cellIndex: Int) = createCell(cellIndex).apply { cellStyle = dateCellStyle }

}